src/main/java/org/gaul/s3proxy/S3ProxyHandlerJetty.java

```diff
@@ -69,10 +80,25 @@ public void handle(String target, Request baseRequest,
             baseRequest.setHandled(true);
             return;
         } catch (HttpResponseException hre) {
-            HttpResponse httpResponse = hre.getResponse();
-            response.sendError(httpResponse == null ?
-                    HttpServletResponse.SC_INTERNAL_SERVER_ERROR :
-                    httpResponse.getStatusCode());
+            HttpResponse hr = hre.getResponse();
+            if (hr == null) {
+                logger.error("Handler failed without response", hre);
```

Comment:
Remove log message.  Thanks to your earlier logging we know the server can just close the stream midway.